File: Frontend/Private/gco-private/src/app/model/core/team-scoreboard-slot.ts
import { SportsOrg } from './sports-org'

export interface TeamScoreboardSlot{
    id: string;
    place: number;
    sportsOrganization: SportsOrg;
    teamNumber: number;
    apparatusTotalScores: Map<string, number>;
    totalScore: number;
}

File: Backend/scoring_service/core/domain/team_scoreboard_slot.go
package domain

import "github.com/google/uuid"

type TeamScoreboardSlot struct {
	ID                   uuid.UUID
	Place                int
	SportsOrganizationID uuid.UUID
	SportsOrganization   SportsOrganization
	TeamNumber           int
	Scores               []Score               `gorm:"-"`
	ApparatusTotalScores map[Apparatus]float32 `gorm:"serializer:json"`
	TotalScore           float32
	TeamScoreboardID     uuid.UUID
	TeamScoreboard       TeamScoreboard
}
